packages/sil-voting-app/src/communication/identities/index.ts

```diff
@@ -0,0 +1,56 @@
+/*global chrome*/
+import { isPublicIdentity, PublicIdentity } from '@iov/bcp';
+import { TransactionEncoder } from '@iov/core';
+import { isJsonRpcErrorResponse, JsonRpcRequest, makeJsonRpcId, parseJsonRpcResponse2 } from '@iov/jsonrpc';
+import { extensionId } from '..';
+
+export const generateGetIdentitiesRequest = (): JsonRpcRequest => ({
+  jsonrpc: '2.0',
+  id: makeJsonRpcId(),
+  method: 'getIdentities',
+  params: {
+    reason: TransactionEncoder.toJson('I would like to know who you are on Ethereum'),
+    chainIds: TransactionEncoder.toJson(['ethereum-eip155-5777']),
+  },
+});
+
+const isArrayOfPublicIdentity = (data: any): data is ReadonlyArray<PublicIdentity> => {
+  if (!Array.isArray(data)) {
+    return false;
+  }
+
+  return data.every(isPublicIdentity);
+};
+
+type GetIdentitiesResponse = ReadonlyArray<PublicIdentity> | undefined;
+
+export const sendGetIdentitiesRequest = async (): Promise<GetIdentitiesResponse> => {
+  const request = generateGetIdentitiesRequest();
+
```

Comment:
Check again my PR you can have problems if you do not check if the variable chrome is available, this might happen if the browser does not have any extension.

Reply:
Addressed in 051d4c71f5ccc759afe99f19c20aeba886220888.